### ISSUE TITLE
Autocorrect on macOS breaks input

### ISSUE THREAD

**Issue by domoritz** [element-hq/element-x-ios#1786]
### Steps to reproduce

I use element X on macOS (Mac with apple silicon) on Sonoma. Try typing a longer message. 

### Outcome

I noticed that whenever I encounter a place where the OS auto-completes a word, the beginning of the word gets removed. This is pretty common and definitely easy to reproduce with any sentence. However, I found that if I type the same word again, the message doesn't break. 

### Your phone model

macBook pro

### Operating system version

macOS Sonoma

### Application version

1.3.0 (72)

### Will you send logs?

No

**Comment by domoritz** [comment]
I don't think this is occasional. For me this happens in almost every single message so that Element X is pretty much useless on Mac. I use Element X on my phone exclusively but on Mac I always use Element because of this bug.

**Comment by pixlwave** [comment]
In this instance, occasional refers to the fact that the overwhelming majority of our users are on iPhone rather than Mac.

**Comment by Kimiblock** [comment]
Somehow the iOS version breaks Chinese (probably CJK) input method.

I’ll later attach a video.

**Comment by domoritz** [comment]
I think this issue is also happening on iOS where if I do a backspace when a recommendations shown, element X keeps the recommendation after the cursor.

**Comment by manuroe** [comment]
Duplicate of https://github.com/element-hq/element-x-ios/issues/2491

**Comment by domoritz** [comment]
It's probably the same underlying bug and since the other issue is marked as higher severity/frequency it makes sense to dupe.

**Comment by stefanceriu** [comment]
Reopening this as per https://github.com/element-hq/element-x-ios/issues/2491#issuecomment-2143197811

**Comment by domoritz** [comment]
In 1.6.13?

**Comment by Velin92** [comment]
it's on develop, so it will be available in the next release

**Comment by domoritz** [comment]
I have  1.7.1 (117) and autocomplete is still broken on macOS. Can you confirm the version this is fixed in?

**Comment by Velin92** [comment]
yes it should

**Comment by domoritz** [comment]
Thanks for looking into this. This is on the latest versions:

https://github.com/user-attachments/assets/db04f344-cdec-4731-966b-06ddfc84861b

**Comment by domoritz** [comment]
@Velin92 can you reopen the issue?

**Comment by Velin92** [comment]
Okay I see, the issue is not in the RTE, but on the plain composer. In the meantime the issue should be fixes if you enable the formatting toolbar which will use the rich text editor composer, we will investigate what is wrong with the plain text one.

**Comment by josephnoir** [comment]
It happens for me in both composers. Using macOS `14.6.1` and  Element X `Version 1.7.3 (119)`.

https://github.com/user-attachments/assets/02b28b2a-db1b-493d-8e70-258a0200bdd6
